Framework/API/inc/MantidAPI/Algorithm.tcc

```diff
@@ -2,7 +2,9 @@
 #include "MantidAPI/IndexProperty.h"
 #include "MantidAPI/WorkspaceProperty.h"
 
+#ifdef _WIN32
 #include "MantidKernel/ArrayProperty.tcc"
```

Comment:
This is the issue was speaking about before. If I remove the include, the build fails in Windows. If I remove the platform guard, this fails in Ubuntu. I tried other solutions to ensure Windows could find the template definitions but this is the only thing that works (see my previous comment on the PR)